### DISCUSSION TITLE
Does OAS3 callbacks also relate about the Service implementation part, because if at all this callback endpoint might be a part of different Swagger as well ?

### DISCUSSION THREAD

**Discussion by mma5997** [OAI/OpenAPI-Specification#2505] [Q&A]
Hi folks,
So my query is, lets say I have OAS3 based Swagger101.json that has a subscription endpoint operation which is configured with a callbacks operation for callbacks URL.

So apart from invocation of callback, does the callbacks definition for this mean that the callbacks URL belongs to some other Service Swagger spec or basically another Swagger JSON. There might be a possibility that this callbacks endpoint operation might be part of whole different Swagger spec that has a lot of operations and endpoints to offer.

Does this make sense ?

Hi @webron , @hkosova ,
any views over this will be much appreciated Thanks :)

**Comment by darrelmiller** [comment]
A URL registered via a subscription endpoint that would later be called using a callback, may very well be described in some other OpenAPI description.  However, I think it would be a bit redundant.  The service that initiates the callback call is the one that defines the shape of the request and the potential response statuses.  The service that listens for the callback has no control over the inbound request and therefore there may not be much value in having a description for it.

**Comment by handrews** [comment]
Answered years ago with no further questions, closing.
